### ISSUE TITLE
New `lint/redundant-call-str` gives false positives

### ISSUE THREAD

**Issue by svdo** [NoahTheDuke/splint#20]
Thanks for this useful new lint rule!

I ran into a false positive though. Given the following file `sample.clj`:

```clojure
(ns sample)

(prn (-> "foo" (str "bar")))
```

Running splint gives:

```
sample.clj:3:16 [lint/redundant-str-call] - Use the literal directly.
(str "bar")
Consider using:
"bar"
```

This is incorrect, `str` is necessary when this occurs in threads (any type of thread of course).

**Comment by NoahTheDuke** [comment]
Fixed in 72add52, release forthcoming.

**Comment by NoahTheDuke** [comment]
Released in [1.17.0](https://github.com/NoahTheDuke/splint/releases/tag/v1.17.0).
